### PR TITLE
[Impeller] Conditionally set command debug info.

### PR DESCRIPTION
**Pull request by jonahwilliams**
If IMPELLER_DEBUG is not defined (release mode), then remove the cmd label strings and additional encoding debug info.

### CODE REVIEW

**Review by jason-simmons — APPROVED** [review]

File: impeller/renderer/command.h [hunk 1]
```diff
@@ -25,6 +25,12 @@
 
 namespace impeller {
 
+#ifdef IMPELLER_DEBUG
+#define DEBUG_COMMAND_INFO(obj, arg) obj.label = arg;
```
Comment (bdero) [inline impeller/renderer/command.h:29]: Prefix with Impeller?

**Review by bdero — COMMENTED** [review]
I recall this coming up before, and we decided to keep the labels at the time. But that was at a time when none of the labels were dynamically generated.

I don't have a strong opinion about this, but @chinmaygarde might have thoughts.

**Comment by chinmaygarde** [comment]
Perhaps we can have something like impeller::DebugLabel or something that turns itself off in certain modes. That way, we won't have to use macros everywhere.

**Comment by jonahwilliams** [comment]
TBH I kind of like the macros because its guaranteed to work, otherwise we get a compile time warning, plus command size is fully shrunk.

We could create a impeller::DebugLabel class that discards its string on construction. Maybe that would be more useful if we want to toggle this on or off?

**Comment by chinmaygarde** [comment]
Yeah, I was thinking of something similar https://godbolt.org/z/cGzWT418G. Also, the debug label could only copy the string out for dynamically allocated strings. When we want absolutely no debug labels, we can ifdef out everything and the struct size is zero.

To be clear, we can do this later. Just want to make sure we are utilizing all tricks we can to ensure we have maximum debuggability. If we get into the habit of getting rid of these labels, we'll find these are too expensive when we need them.

**Review by chinmaygarde — APPROVED** [review]
Just LGTMing this now to unblock. We can rework debug labels later at any point.

**Pull request merged**